particle-iot/device-os
# [esomx] map ADC channels to the correct pins

## `NOTE: This PR should also be backported to 4.x`

### Problem

ESoMX ADC pins were mapped incorrectly, which caused `analogRead(A0~A7)` to fail.

### Solution

Remap the ADC channel/pins per Boron/BSoM mapping.

### Steps to Test

- Run the example app below and look for correct behavior with a 10k potentiometer connected to 3.3V and GND, wiper on the pin under test.  Set the pot to about halfway.  This should be about 5k ohms of impedance and give a stable reading.  Walk the connection between A0 - A7 as the test advances.  The reading should be stable.
- Run `TEST=wiring/no_fixture` and `GPIO_08_DigitalReadWorksMixedWithAnalogRead` should pass.

### Example App

```c
#include "Particle.h"
SYSTEM_MODE(SEMI_AUTOMATIC);
SYSTEM_THREAD(ENABLED);
Serial1LogHandler log1Handler(115200, LOG_LEVEL_ALL);
SerialLogHandler logHandler(LOG_LEVEL_ALL);

void setup() {
    for (int i = 10; i < 18; i++) {
        for (int x = 0; x < 10; x++) {
            Log.info("Reading (A%d): %ld", i-10, analogRead(i));
            delay(1000);
        }
    }
}
void loop() {
}
```

---

### Completeness
